lib/token-persistence.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const autosave = require('json-autosave')
+const githubAuth = require('./github-auth')
+
+// This is currently bound to the legacy github auth code. That will be
+// replaced with a dependency-injected token provider.
+class TokenPersistence {
+  constructor({ path }) {
+    this.path = path
+    this.save = undefined
+  }
+
+  async initialize() {
```

Comment:
This code is a bit difficult to understand, in part because it's working against the interface of `json-autosave` which wants to own the data structure.

Reply:
Not trivial indeed, but I think I get the gist of it. 👍